internal/jobcontainers/logon.go

```diff
@@ -0,0 +1,62 @@
+package jobcontainers
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Microsoft/hcsshim/internal/winapi"
+	"github.com/pkg/errors"
+	"golang.org/x/sys/windows"
+)
+
+// processToken returns a user token for the user specified by `user`. This should be in the form
+// of either a DOMAIN\username or just username. `inherit` is a bool that signifies to
+// just return (inherit) the token from the parent process.
+func processToken(user string, inherit bool) (windows.Token, error) {
+	var (
+		domain   string
+		userName string
+		token    windows.Token
+	)
+
+	split := strings.Split(user, "\\")
+	if len(split) == 2 {
+		domain = split[0]
+		userName = split[1]
+	} else if len(split) == 1 {
+		userName = split[0]
+	} else {
+		return 0, fmt.Errorf("invalid user string `%s`", user)
+	}
+
+	// If empty just let it inherit the token from whatever is used to launch it (containerd-shim etc).
+	if user == "" || inherit {
+		return openCurrentProcessToken()
```

Comment:
I don't think we should use the current process token if `user` is empty, that should be an error condition unless `inherit` is true.

I also think we should just simplify this function by removing `inherit` from it and just have the caller decide whether to call `processToken` or `openCurrentProcessToken` based on the annotation value.

Reply:
Good point I forgot to remove the empty string case, fixing